examples/disclosure/js/disclosure-button.js

```diff
@@ -0,0 +1,102 @@
+/*
+ *   This content is licensed according to the W3C Software License at
+ *   https://www.w3.org/Consortium/Legal/2015/copyright-software-and-document
+ *
+ *   File:   disclosure-button.js
+ *
+ *   Desc:   Disclosure button widget that implements ARIA Authoring Best Practices
+ */
+
+'use strict';
+
+/*
+ *   @constructorDisclosureButton
+ *
+ *
+ */
+class DisclosureButton {
+  constructor(buttonNode) {
+    this.buttonNode = buttonNode;
+    this.controlledNode = false;
+
+    var id = this.buttonNode.getAttribute('aria-controls');
+
+    if (id) {
+      this.controlledNode = document.getElementById(id);
+    }
+
+    this.buttonNode.setAttribute('aria-expanded', 'false');
+    this.hideContent();
+
+    this.buttonNode.addEventListener('keydown', this.onKeydown.bind(this));
+    this.buttonNode.addEventListener('click', this.onClick.bind(this));
+    this.buttonNode.addEventListener('focus', this.onFocus.bind(this));
+    this.buttonNode.addEventListener('blur', this.onBlur.bind(this));
+  }
+
+  showContent() {
+    if (this.controlledNode) {
+      this.controlledNode.style.display = 'block';
+    }
+  }
+
+  hideContent() {
+    if (this.controlledNode) {
+      this.controlledNode.style.display = 'none';
+    }
+  }
+
+  toggleExpand() {
+    if (this.buttonNode.getAttribute('aria-expanded') === 'true') {
+      this.buttonNode.setAttribute('aria-expanded', 'false');
+      this.hideContent();
+    } else {
+      this.buttonNode.setAttribute('aria-expanded', 'true');
+      this.showContent();
+    }
+  }
+
+  /* EVENT HANDLERS */
+
+  onKeydown(event) {
+    switch (event.keyCode) {
+      case this.keyCode.RETURN:
```

Comment:
Quick question -- why do we need to handle a return key press? Since it's a button element, shouldn't using the click event work fine?

Reply:
I agree, the `click` event is input device agnostic for elements with an activation behavior, including `button`: https://w3c.github.io/uievents/#event-flow-activation

Reply:
@zcorpan @smhigley I removed the keyboard event handler, everything works the same.